### PR TITLE
Restart enable touchups

### PR DESCRIPTION
**Pull request by jrvollmer**
Touched up appearance and contents of restart enable popup
- Shows active faults in red, previously active faults in white, and faults that haven't been triggered in grey
- More defined layout (boxes containing faults)
- Contactors added to faults
- Faults set to nominal values (T/F) initially to avoid opening the restart enable popup before any data is generated/retrieved

Fixed fan speed warnings (see UI/BatterySolarBox.qml)
Updated speed to float in UI, DataGen, and DataUnpacker to match data format

### CODE REVIEW

File: DataProcessor/DataGen.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 #include "DataGen.h"
 
-int lastSpeed=0;
+float lastSpeed=0;
```
Comment (EUdds) [inline DataProcessor/DataGen.cpp]: Lets avoid global variables wherever possible

File: DataProcessor/DataGen.cpp [hunk 2]
```diff
@@ -28,22 +28,19 @@ void DataGen::getData(QByteArray &data, std::vector<std::string> &names, std::ve
                 addFloatToArray(data,(float)batteryFunc(timeArg));
             } else if(names[i] == "accelerator") {
                 addFloatToArray(data,std::fmod(timeArg,5)/*(float)rand()/((RAND_MAX+1u)/5)*/);
+            } else if(names[i] == "speed") {
```
Comment (EUdds) [inline DataProcessor/DataGen.cpp:27]: Future ticket - but this if/else tree is getting mighty long, we should look at more efficient ways to do this

File: DataProcessor/DataUnpacker.cpp [hunk 3]
```diff
@@ -149,7 +149,7 @@ void DataUnpacker::unpack()
     
     mutex.unlock();
 
-    this->restart_enable = !battery_eStop || !driver_eStop || !external_eStop || !imd_status || !door || crash || !mcu_check || restart_enable;
+    this->restart_enable = battery_eStop || driver_eStop || external_eStop || imd_status || !door || crash || mcu_check || restart_enable;
```
Comment (EUdds) [inline DataProcessor/DataUnpacker.cpp]: This also do be mighty long - possible place for a function

File: DataProcessor/DataUnpacker.h [hunk 4]
```diff
@@ -87,14 +90,16 @@ public slots:
...
     QThread dataHandlingThread;
 
-    // TOOD Include only the properties that need to be displayed on the driver dashboard
-    uint8_t speed, fan_speed, tstamp_hr, tstamp_mn, tstamp_sc;
+    // TODO Include only the properties that need to be displayed on the driver dashboard
+    uint8_t fan_speed, tstamp_hr, tstamp_mn, tstamp_sc;
     uint16_t tstamp_ms;
-    float accelerator, soc, mppt_current_out, pack_voltage, pack_current, pack_temp, motor_temp, driverIO_temp, mainIO_temp, cabin_temp, string1_temp, string2_temp, string3_temp;
-    bool headlights, cruise, left_turn, right_turn, hazards, mainIO_heartbeat;
+    float speed, accelerator, soc, mppt_current_out, pack_voltage, pack_current, pack_temp, motor_temp, driverIO_temp, mainIO_temp, cabin_temp, string1_temp, string2_temp, string3_temp;
```
Comment (EUdds) [inline DataProcessor/DataUnpacker.h]:
```suggestion
    float speed, accelerator;
    float soc;
    float mppt_current_out;
    float pack_voltage, pack_current;
    float pack_temp, motor_temp, driverIO_temp, mainIO_temp, cabin_temp;
    float string1_temp, string2_temp, string3_temp;
```

File: DataProcessor/DataUnpacker.h [hunk 5]
```diff
@@ -87,14 +90,16 @@ public slots:
...
+    bool headlights, cruise, left_turn, right_turn, hazards, mainIO_heartbeat, eng_dash_commfail=1;
```
Comment (EUdds) [inline DataProcessor/DataUnpacker.h]:
```suggestion
    bool headlights, cruise, left_turn, right_turn, hazards, mainIO_heartbeat;
    bool eng_dash_commfail=1;
```

**Review by EUdds — CHANGES_REQUESTED** [review]

File: DataProcessor/DataGen.cpp (hunk 2 above)
Comment (jrvollmer) [reply DataProcessor/DataGen.cpp:27]: Ticket for improving DataGen has been added to Jira

File: DataProcessor/DataGen.cpp [hunk 6]
```diff
@@ -28,36 +24,34 @@ void DataGen::getData(QByteArray &data, std::vector<std::string> &names, std::ve
...
         } else if(types[i] == "bool") {
             // For shutdown circuit inputs, any triggerred error will stay triggered for 3 seconds after the most recent error is triggered
             // When a new error is triggerred, the countdown will restart from 3 seconds for all currently triggered errors
-            if((names[i] == "battery_eStop") || (names[i] == "driver_eStop") || (names[i] == "external_eStop") ||
-               (names[i] == "imd_status") || (names[i] == "door") || (names[i] == "mcu_check")) {
+            if(names[i] == "door") {
                 // NC/preferred true shutdown circuit inputs
                 std::size_t errPos = errors.find(names[i]);
-                bool error = (rand()%200+1 >= 2) && !((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
+                bool error = (rand()%700+1 >= 2) && !((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
```
Comment (EUdds) [inline DataProcessor/DataGen.cpp]: Let's remove magic numbers or at least make a function to tell me what is going on here

File: DataProcessor/DataGen.cpp [hunk 7]
```diff
@@ -28,36 +24,34 @@ void DataGen::getData(QByteArray &data, std::vector<std::string> &names, std::ve
...
                 } else if((errPos != std::string::npos) && (errStartTime <= time(NULL) - 3)) {
                     errors.erase(errPos,names[i].length());
                 }
-            } else if(names[i] == "crash") {
+            } else if((names[i] == "battery_eStop") || (names[i] == "driver_eStop") || (names[i] == "external_eStop") ||
+                      (names[i] == "imd_status") || (names[i] == "crash") || (names[i] == "mcu_check")) {
                 // NO/preferred false shutdown circuit inputs
                 std::size_t errPos = errors.find(names[i]);
-                bool error = (rand()%200+1 <= 2) || ((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
+                bool error = (rand()%700+1 <= 2) || ((errStartTime > time(NULL) - 3) && (errPos != std::string::npos));
```
Comment (EUdds) [inline DataProcessor/DataGen.cpp]: here too

**Review by EUdds — CHANGES_REQUESTED** [review]
Almost perfect

**Review by EUdds — APPROVED** [review]

**Pull request merged**